PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/controllers/api/ImageController.java

```diff
@@ -0,0 +1,58 @@
+package pl.polsl.photoplus.controllers.api;
+
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import pl.polsl.photoplus.annotations.validators.Image;
+import pl.polsl.photoplus.model.dto.ImageModelDto;
+import pl.polsl.photoplus.services.controllers.ImageService;
+
+import javax.validation.Valid;
+import java.io.IOException;
+import java.util.List;
+
+@Validated
+@RestController
+@RequestMapping("/image")
+public class ImageController {
+
+    private ImageService imageService;
+
+    public ImageController(final ImageService imageService) {
+        this.imageService = imageService;
+    }
+
+    @GetMapping(path = "/{code}", produces = MediaType.IMAGE_PNG_VALUE)
+    @PreAuthorize("hasPermission('image', 'single' )")
+    public ResponseEntity<ByteArrayResource> getSingle(@PathVariable("code") final String code) {
+        final ImageModelDto img = imageService.getSingleObject(code);
```

Comment:
Does it make sense to create Dto? Maybe passing image directly would be better if dto is not returned to user?